test/integration/resource_test.go

```diff
@@ -203,51 +220,3 @@ func TestUpdateResourceWithRacingRequests(t *testing.T) {
 	// the resource patch request is protected by the advisory lock, so there should only be one update
 	Expect(updatedCount).To(Equal(1))
 }
-
-func TestUpdateResourceWithRacingRequests_WithoutLock(t *testing.T) {
```

Comment:
I don't think we want to remove this.

Your test is about version conflicts while this test is about concurrency and locking. They are different.